src/cryptenroll/cryptenroll.c

```diff
@@ -28,6 +28,7 @@
 #include "tpm2-util.h"
 
 static EnrollType arg_enroll_type = _ENROLL_TYPE_INVALID;
+static char *arg_unlock_file = NULL;
```

Comment:
hmm, so crypttab and cryptsetup call the concept "keyfile". We should probably stick to the same naming, to minimize surprises.

Reply:
Just to be sure. In https://github.com/systemd/systemd/issues/23295#issuecomment-1122119881 you said `--unlock-file=` make sense to you.
So are we using `--unlock-file=` and `--key-file=` or `--key-file=` and an unnamed parameter with the keyfile path like `cryptsetup`?

Reply:
uh, oh. thanks for pointing me to my own earlier contradicting words ;-)

Hmm, so I think my current self would agree with my younger self there, i.e. that this probably should be called something with "unlock" to indicate that this is about unlocking for enrollment, not for enrolling. But my current self also has a point, it should use existing nomenclature. So maybe the combination of both make most sense, i.e. `--unlock-key-file=…`. That way we have the best of both worlds, and call it "key file" but also emphasize it is about "unlocking".

Does that make sense?

Reply:
So this isn't your usual keyfile (e.g. binary blob) we are used to from cryptsetup operations, but rather a file containing the password (which I think is restricted in content to some degree?), right? In that case it should really be called password_file or something like that to drive that point home.

Reply:
Hmm, it isn't you usual key file? huh?

Reply:
My cursory glance at this was that this option is just a means to pass the password to sd-cryptentroll (non-interactively). Whereas a keyfile would be a binary blob and something else entirely. I might be wrong here, though.

Reply:
cryptsetup/LUKS do not distinguish passwords from binary blobs used to unlock stuff. It's always just some data with a size...

Reply:
I would beg to differ (yes, the low-level stuff doesn't care, but the UI should/does): https://man.archlinux.org/man/cryptsetup.8#Passphrase_processing_for_LUKS

Reply:
I don't grok what you are trying to say?

Reply:
> So maybe the combination of both make most sense, i.e. `--unlock-key-file=…`. That way we have the best of both worlds, and call it "key file" but also emphasize it is about "unlocking".

This sounds like an good idea.